src/routes/transactions/mappers/common/transactions.mapper.ts

```diff
@@ -0,0 +1,205 @@
+    if (transactionsDomain.length == 0) {
+      return [];
+    }
+    const prevPageTimestamp = this.getPreviousPageDate(
+      transactionsDomain,
+      offset,
+      timezoneOffset,
+    )?.getTime();
+    // Remove first item that was requested to get previous day timestamp
+    if (prevPageTimestamp !== undefined) {
+      transactionsDomain = transactionsDomain.slice(1);
+    }
+    const transactionsDomainGroups = this.groupByDay(
+      transactionsDomain,
+      timezoneOffset,
+    );
+
+    const transactionList = await Promise.all(
+      transactionsDomainGroups.map(async (transactionGroup) => {
+        const transactions: (TransactionItem | DateLabel)[] = [];
+        if (transactionGroup.timestamp != prevPageTimestamp) {
+          transactions.push(new DateLabel(transactionGroup.timestamp));
+        }
+        transactions.push(
+          ...(await this.mapGroupTransactions(transactionGroup, chainId, safe)),
+        );
+        return transactions.flat();
+      }),
+    );
+
+    return transactionList.flat();
+  }
+
+  private getTransactionTimestamp(transaction: TransactionDomain): Date {
+    let timestamp: Date | null;
+    if (isMultisigTransaction(transaction)) {
+      const executionDate = transaction.executionDate;
+      timestamp = executionDate ?? transaction.submissionDate;
+    } else if (isEthereumTransaction(transaction)) {
+      timestamp = transaction.executionDate;
+    } else if (isModuleTransaction(transaction)) {
+      timestamp = transaction.executionDate;
+    } else if (isCreationTransaction(transaction)) {
+      timestamp = transaction.created;
+    } else {
+      throw Error('Unknown transaction type');
+    }
+    if (timestamp == null) {
+      throw Error('ExecutionDate cannot be null');
+    }
+    return timestamp;
+  }
+
+  private getPreviousPageDate(
+    transactions: TransactionDomain[],
+    offset: number,
+    timezoneOffset: number,
+  ): Date | undefined {
+    // More than 1 element is required to get the previous page date
+    if (offset <= 0 || transactions.length <= 1) return undefined;
+    const previousPageTransaction = transactions[0];
+    const timestamp = this.getTransactionTimestamp(previousPageTransaction);
+    return this.getDayFromDate(timestamp, timezoneOffset);
+  }
+
+  private groupByDay(
+    transactions: TransactionDomain[],
+    timezoneOffset: number,
+  ): TransactionDomainGroup[] {
+    return Object.entries(
+      groupBy(transactions, (transaction) => {
+        const transactionTimestamp = this.getTransactionTimestamp(transaction);
+        return this.getDayFromDate(
+          transactionTimestamp,
+          timezoneOffset,
+        ).getTime();
+      }),
+    ).map(
+      ([timestamp, transactions]) =>
+        <TransactionDomainGroup>{
+          timestamp: Number(timestamp),
+          transactions: transactions,
+        },
+    );
+  }
+
+  private getDayFromDate(timestamp: Date, timezoneOffset: number): Date {
```

Comment:
I'd add documentation to this function. I think it's not trivial to know its expected output and the role of `timezoneOffset` on it 🙂

Reply:
Done 14bf055fda1872f7fabc8b04cf92820621f778f3